Bluetooth: Controller: Fix missing dup filtering cond. compile

cvinayak opened the pull request:
Fix missing duplicate filtering feature conditional
compilation which caused build error when duplicate
filtering feature is disabled when Periodic Advertising
feature selected.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Fix missing dup filtering cond. compile

Fix missing duplicate filtering feature conditional
compilation which caused build error when duplicate
filtering feature is disabled when Periodic Advertising
feature selected.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Thalley reviewed (dismissed):
Can we test things like this in the future somehow? 

e.g. for LE Audio we've created tests that just builds with different configurations to (for the most part) ensure that enabling/disabling Kconfigs does not cause build issues.

cvinayak committed:
tests: Bluetooth: init: Periodic Advertising ADI support

Add build only tests to cover Periodic Advertising ADI
support in advertising state and in Periodic
Synchronization state.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak commented:
> Can we test things like this in the future somehow?
> 
> e.g. for LE Audio we've created tests that just builds with different configurations to (for the most part) ensure that enabling/disabling Kconfigs does not cause build issues.

Controller does have build only tests for most top level features, add tests to cover ADI support now in this PR
